simapp/helpers/test_helpers.go

```diff
@@ -14,7 +14,7 @@ import (
 
 // SimAppChainID hardcoded chainID for simulation
 const (
-	DefaultGenTxGas = 1000000
+	DefaultGenTxGas = 4000000 // An increase is needed when adding checks to staking/msg_server.go
```

Comment:
a 2-4x increase in gas for couple fields and few checks doesn't look so right to me. What do you think?